### ISSUE TITLE
How to turn Rust Image buffer Vec<u8> to JsBuffer?

### ISSUE THREAD

**Issue by WeismannS** [neon-bindings/neon#980]
Code snipper:
I'm using the `image` and `imageproc` crates
```rust
let buffer =  image3.into_raw();
 Ok(cx.buffer(buffer as usize)?)``` obviously doesn't work but i don't what to do

**Comment by kjvalencik** [comment]
https://docs.rs/neon/1.0.0-alpha.2/neon/types/buffer/trait.TypedArray.html#tymethod.from_slice

```rust
use neon::types::JsUint8Array;

// Replace with the real data in a `Vec<u8>`
let data: Vec<u8> = vec![1, 2, 3, 4];

JsUint8Array::from_slice(&data)
```

Note, as of writing this is only available in the 1.0 pre-release (1.0.0-alpha.2). In `0.10`, there is a bit more boilerplate.

```rust
let mut buf = cx.array_buffer(data.len());

buf.as_mut_slice(&mut cx).copy_from_slice(&data);

Ok(buf)
```

**Comment by WeismannS** [comment]
> https://docs.rs/neon/1.0.0-alpha.2/neon/types/buffer/trait.TypedArray.html#tymethod.from_slice
> 
> ```rust
> use neon::types::JsUint8Array;
> 
> // Replace with the real data in a `Vec<u8>`
> let data: Vec<u8> = vec![1, 2, 3, 4];
> 
> JsUint8Array::from_slice(&data)
> ```
> 
> Note, as of writing this is only available in the 1.0 pre-release (1.0.0-alpha.2). In `0.10`, there is a bit more boilerplate.
> 
> ```rust
> let mut buf = cx.array_buffer(data.len());
> 
> buf.as_mut_slice(&mut cx).copy_from_slice(&data);
> 
> Ok(buf)
> ```

Thank you! so there is no way of doing it on the latest stable version?

**Comment by WeismannS** [comment]
following your code for 0.10

> https://docs.rs/neon/1.0.0-alpha.2/neon/types/buffer/trait.TypedArray.html#tymethod.from_slice
> 
> ```rust
> use neon::types::JsUint8Array;
> 
> // Replace with the real data in a `Vec<u8>`
> let data: Vec<u8> = vec![1, 2, 3, 4];
> 
> JsUint8Array::from_slice(&data)
> ```
> 
> Note, as of writing this is only available in the 1.0 pre-release (1.0.0-alpha.2). In `0.10`, there is a bit more boilerplate.
> 
> ```rust
> let mut buf = cx.array_buffer(data.len());
> 
> buf.as_mut_slice(&mut cx).copy_from_slice(&data);
> 
> Ok(buf)
> ```
```console
error [E0599]: no method named `as_mut_slice` found for struct `Handle<'_, JsArrayBuffer>` in the current scope
  --> src\lib.rs:37:9
   |
37 |     buf.as_mut_slice(&mut cx).copy_from_slice(&data)?;
   |         ^^^^^^^^^^^^ method not found in `Handle<'_, JsArrayBuffer>`
   |
```
0.10

**Comment by kjvalencik** [comment]
As linked above, that method is on the `TypedArray` trait which needs to be in scope to call it.

```
use neon::types::buffer::TypedArray;
```

I hope this helps. I'm going to close this issue since it's more of a support request than an issue with Neon. Cheers!
